### PR TITLE
Set up basic Docusaurus files, add Actions workflow for deploying, Dependabot config

### PR DESCRIPTION
**Pull request by jodyheavener**
This PR adds the initial setup files for our Experimenter documentation hub, using [Docusaurus](https://v2.docusaurus.io/).

I did very little in the way of customizing the theme or adding content, but I did:

- Setup basic names and descriptions in the config file
- Add an Experimenter "logo"
- Write a very basic README for setting up and contributing

Also, and somewhat importantly, I have added a proposed way for us to deploy changes:

- Using the recommended [GitHub Actions method of deploying](https://v2.docusaurus.io/docs/deployment#triggering-deployment-with-github-actions), I added an Actions workflow file
  - I was originally going to copy the way our friends over at FXA are doing it in [ecosystem-platform](https://github.com/mozilla/ecosystem-platform/blob/master/.circleci/config.yml) by using CircleCI to deploy, but we have Actions now so if we don't need to introduce an external tool I'd rather not.
- The workflow is more or less a direct copy-paste from Docusaurus, but I changed the watched branch to `main` as this repo is dedicated to docs.
- The repo **currently** has the necessary [Deploy Key](https://github.com/mozilla/experimenter-docs/settings/keys) (public key) and [Secret](https://github.com/mozilla/experimenter-docs/settings/secrets/actions) (private key) set up to facilitate this deploy (I hope, won't know until it's deployed).
- Once the first deploy is ready we can change the repo settings to point GitHub Pages at the `gh-pages` branch.

Finally, also added Dependabot config.

### CODE REVIEW

File: README.md
Comment (lmorchard) [inline README.md:7]: `s/Pages is/Pages are/`

File: README.md
Comment (lmorchard) [inline README.md:13]: We probably want mention & suggestion to submit PRs from edits? Don't think we'll want folks working directly on main branch.

Been awhile since I did PRs purely from the github web UI, but I think you can edit markdown in-browser to create PRs

File: .github/workflows/documentation.yml
Comment (lmorchard) [inline .github/workflows/documentation.yml:37]: I think we're [missing a few conditionals](https://v2.docusaurus.io/docs/deployment#triggering-deployment-with-github-actions) in this script and it's [causing an error](https://github.com/mozilla/experimenter-docs/runs/2008160447?check_suite_focus=true). (Or at least, the "check" job failed on this, but I don't see that in this action any more)

Looks like it needs the checks for yarn.lock vs package-lock.json to run just yarn or npm as needed:
```
          if [ -e yarn.lock ]; then
          yarn install --frozen-lockfile
          elif [ -e package-lock.json ]; then
          npm ci
          else
          npm i
          fi
          npx docusaurus deploy
```

**Review by lmorchard — APPROVED** [review]
I think there's a problem in the gh-release action's script that will error out. Looks like there might have been a `check` action before, which [failed over here](https://github.com/mozilla/experimenter-docs/runs/2008160447?check_suite_focus=true) for the same reason.

Not going to block merge, though, since this CI stuff can be hard to debug without trying it first and this is just the initial setup PR. Probably want to tweak that script before merge though

File: .github/workflows/documentation.yml
Comment (jodyheavener) [reply .github/workflows/documentation.yml:37]: :hehehe: Whoops, I thought I had remove the NPM stuff since we are using Yarn, guess I missed some. Thanks for catching!

File: README.md
Comment (jodyheavener) [reply README.md:13]: Yeah this write-up definitely needs some finessing - this is kind of just an initial go at it. I want to go through the flow (of creating new and editing existing MD files) myself and then I can more accurately update this.

File: .github/workflows/documentation.yml
Comment (jodyheavener) [reply .github/workflows/documentation.yml:37]: Oh also, there used to be a checks workflow that I removed, that's what that failing workflow you linked to is from

File: README.md
Comment (LZoog) [reply README.md:13]: FWIW, we'll be writing "official" docs on this in EXP-1027

File: .github/workflows/documentation.yml
Comment (lmorchard) [reply .github/workflows/documentation.yml]: Yeah, we might actually want that checks workflow just to sanity check PRs since there is a build process here

**Review by LZoog — APPROVED** [review]
👍

We probably want to open a dependabot here too perhaps? Could be worth filing an issue for

**Comment by jodyheavener** [comment]
Hell ya deploy worked first try 🥳

**Pull request merged**
